repos/spack_repo/builtin/packages/hypre/package.py

```diff
@@ -59,7 +66,8 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     version("2.10.1", sha256="a4a9df645ebdc11e86221b794b276d1e17974887ead161d5050aaf0b43bb183a")
     version("2.10.0b", sha256="b55dbdc692afe5a00490d1ea1c38dd908dae244f7bdd7faaf711680059824c11")
 
-    variant("shared", default=True, description="Build shared library (disables static library)")
+    variant("shared", default=False, description="Build shared library (disables static library)")
+    variant("pic", default=True, description="Build position independent code")
```

Comment:
Here I would do something similar to:
```suggestion
    variant("pic", default=True, when="~shared", description="Build position independent code")
```
since `+shared~pic` doesn't make much sense.